security/pfSense-pkg-zeek/files/usr/local/pkg/zeek.inc

```diff
@@ -0,0 +1,982 @@
+	 if (!file_exists(ZEEK_ETC . "/zeekctl.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/zeekctl.cfg.backup");
+		 copy(ZEEK_ETC . "/zeekctl.cfg", ZEEK_ETC . "/zeekctl.cfg.backup");
+	 }
+
+	 // creating a backup file of the original /etc/node no matter if user checked this or not
+	 if (!file_exists(ZEEK_ETC . "/node.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/node.cfg.backup");
+		 copy(ZEEK_ETC . "/node.cfg", ZEEK_ETC . "/node.cfg.backup");
+	 }
+
+	 // creating a backup file of the original /etc/networks no matter if user checked this or not
+	 if (!file_exists(ZEEK_ETC . "/networks.cfg.backup")) {
+		 log_error("[zeek] Creating backup of the original file to " . ZEEK_ETC . "/networks.cfg.backup");
+		 copy(ZEEK_ETC . "/networks.cfg", ZEEK_ETC . "/networks.cfg.backup");
+	 }
+	 // Change owner of zeek created files
+	 exec("chown -R root:wheel /usr/local/logs");
+
+	 exec("chown -R root:wheel /usr/local/spool");
+
+	 unlink_if_exists("/usr/local/etc/rc.d/zeek.sh");
+
+	 // create zeek rcfile
+	 zeek_write_rcfile();
+
+	 write_config("[zeek] Package installed.");
+ }
+
+
+ function zeek_deinstall_command() {
+	 zeek_install_cron(false);
+	 mwexec("/bin/ps awux | /usr/bin/egrep -i 'zeek -f|\(zeek\)' | /usr/bin/awk '{ print $2 }' | /usr/bin/xargs kill");
+	 unlink_if_exists("/usr/local/etc/rc.d/zeek.sh");
+
+	 if (zeek_log_resync()) {
+		 if (is_dir("/usr/local/logs")) {
+			 exec("/bin/rm -rf /usr/local/logs/*");
+		 }
+	 }
+
+	 if (is_array($config['installedpackages']['zeek'])) {
+		 unset($config['installedpackages']['zeek']);
+	 }
+	 if (is_array($config['installedpackages']['zeekcontrol'])) {
+		 unset($config['installedpackages']['zeekcontrol']);
+	 }
+	 if (is_array($config['installedpackages']['zeekcluster'])) {
+		 unset($config['installedpackages']['zeekcluster']);
+	 }
+	 if (is_array($config['installedpackages']['zeekscript'])) {
+		 unset($config['installedpackages']['zeekscript']);
+	 }
+	 if (is_array($config['installedpackages']['zeeklog'])) {
+		 unset($config['installedpackages']['zeeklog']);
+	 }
+
+	 if (is_array($config['installedpackages']['zeeksync'])) {
+		 unset($config['installedpackages']['zeeksync']);
+	 }
+ }
+
+ /*
+ * Zeek input validation
+ */
+
+ // Zeek NSM: General Settings input validation
+ function zeek_validate_general($post, &$input_errors) {
+	 global $config;
+
+	 if (is_array($config['installedpackages']['zeek'])) {
+		 $settings = $config['installedpackages']['zeek']['config'][0];
+	 } else {
+		 $settings = array();
+	 }
+
+	 if ($post['enable_zeek'] == "on") {
+		 if (!zeek_cluster_enabled()) {
+			 if (empty($post['active_interface'])) {
+				 $input_errors[] = "You must select at least one interface under 'Zeek Interface(s)' to enable Zeek instance.";
+				 $input_errors[] = "If you intend to use Zeek as cluster, then visit Zeek NSM: Zeek cluster, configure and save the cluster settings first.";
```

Comment:
Reword to avoid addressing the user directly ("You"). Here and in all the other messages below.

https://docs.netgate.com/pfsense/en/latest/references/style-guide.html